internal/rootcoord/meta_table.go

```diff
@@ -1456,3 +1461,118 @@ func (mt *MetaTable) RestoreRBAC(ctx context.Context, tenant string, meta *milvu
+	privGroups, err := mt.catalog.ListPrivilegeGroups(mt.ctx)
+	if err != nil {
+		return false, err
+	}
+	for _, group := range privGroups {
+		if group.GroupName == groupName {
+			return true, nil
+		}
+	}
+	return false, nil
+}
+
+func (mt *MetaTable) CreatePrivilegeGroup(groupName string) error {
+	if funcutil.IsEmptyString(groupName) {
+		return fmt.Errorf("the privilege group name is empty")
+	}
+	mt.permissionLock.Lock()
+	defer mt.permissionLock.Unlock()
+
+	definedByUsers, err := mt.IsCustomPrivilegeGroup(groupName)
+	if err != nil {
+		return err
+	}
+	if definedByUsers {
+		return merr.WrapErrParameterInvalidMsg("privilege group name [%s] is defined by users", groupName)
+	}
+	if util.IsPrivilegeNameDefined(groupName) {
+		return merr.WrapErrParameterInvalidMsg("privilege group name [%s] is defined by built in privileges or privilege groups in system", groupName)
+	}
+	privileges := make([]*milvuspb.PrivilegeEntity, 0)
+	return mt.catalog.AlterPrivilegeGroup(mt.ctx, groupName, privileges)
+}
+
+func (mt *MetaTable) DropPrivilegeGroup(groupName string) error {
+	if funcutil.IsEmptyString(groupName) {
+		return fmt.Errorf("the privilege group name is empty")
+	}
+	mt.permissionLock.Lock()
+	defer mt.permissionLock.Unlock()
+
+	definedByUsers, err := mt.IsCustomPrivilegeGroup(groupName)
+	if err != nil {
+		return err
+	}
+	if !definedByUsers {
+		return merr.WrapErrParameterInvalidMsg("there is no privilege group name [%s] to drop", groupName)
+	}
+	if util.IsPrivilegeNameDefined(groupName) {
+		return merr.WrapErrParameterInvalidMsg("can not drop privilege group name [%s] which is defined by built in privileges or privilege groups in system", groupName)
+	}
+	return mt.catalog.DropPrivilegeGroup(mt.ctx, groupName)
+}
+
+func (mt *MetaTable) ListPrivilegeGroups() ([]*milvuspb.PrivilegeGroupInfo, error) {
+	mt.permissionLock.Lock()
+	defer mt.permissionLock.Unlock()
+
+	return mt.catalog.ListPrivilegeGroups(mt.ctx)
+}
+
+func (mt *MetaTable) OperatePrivilegeGroup(groupName string, privileges []*milvuspb.PrivilegeEntity, operateType milvuspb.OperatePrivilegeGroupType) error {
+	if funcutil.IsEmptyString(groupName) {
+		return fmt.Errorf("the privilege group name is empty")
+	}
+	mt.permissionLock.Lock()
+	defer mt.permissionLock.Unlock()
+
+	// validate input params
+	definedByUsers, err := mt.IsCustomPrivilegeGroup(groupName)
+	if err != nil {
+		return err
+	}
+	if !definedByUsers {
+		return merr.WrapErrParameterInvalidMsg("there is no privilege group name [%s] to operate", groupName)
+	}
+	if util.IsPrivilegeNameDefined(groupName) {
+		return merr.WrapErrParameterInvalidMsg("can not operate privilege group [%s] which is defined by built in privileges or privilege groups in system", groupName)
+	}
+	for _, p := range privileges {
+		if !util.IsPrivilegeNameDefined(p.Name) {
+			return merr.WrapErrParameterInvalidMsg("there is no privilege name [%s] defined in system to operate", p.Name)
```

Comment:
consider a case, if user pass a privilege group in privileges, those err msg will be confused